src/java/main/org/apache/zookeeper/server/NettyServerCnxn.java

```diff
@@ -267,10 +267,17 @@ private boolean checkFourLetterWord(final Channel channel,
     {
         // We take advantage of the limited size of the length to look
         // for cmds. They are all 4-bytes which fits inside of an int
-        String cmd = FourLetterCommands.getCmdMapView().get(len);
-        if (cmd == null) {
+        if (!FourLetterCommands.isKnown(len)) {
             return false;
         }
+
+        // ZOOKEEPER-2693: don't execute 4lw if it's not enabled.
+        String cmd = FourLetterCommands.getCommandString(len);
+        if (!FourLetterCommands.isEnabled(cmd)) {
+            LOG.debug("Command {} is not executed because it is not white listed.", cmd);
+            return true;
```

Comment:
@rakeshadr Turns out server socket not closed is a by design. Had a chat with @phunt offline and the idea is we always prefer client to close socket first (which led to server socket close) because a socket close at server might be premature and led to client not getting all the data due to how TCP works. I'll leave these code as they are.